In [1]:
import pandas as pd
from scipy import stats
import statsmodels.formula.api as smf
import statsmodels.api as sm

In [9]:
df = pd.read_csv("/content/teachers_rating_data (4).csv")
df.head()

prof  Gender  Age Tenure  Evaluation  Students  Beauty CourseLevel
0  Prof49  Female   42    Yes        3.45        49     4.7       Lower
1  Prof97  Female   28    Yes        3.91        55     2.9       Upper
2  Prof12  Female   41     No        4.67        43     3.9       Lower
3  Prof34  Female   41    Yes        4.23        49     4.2       Upper
4  Prof95  Female   28    Yes        3.20        76     4.6       Lower

# Q1. Regression with T-test: Using the teachers rating data set, does gender affect teaching evaluation rates?

In [10]:
model = smf.ols('Evaluation ~ Gender', data=df).fit()
print(model.summary())

                            OLS Regression Results                            
Dep. Variable:             Evaluation   R-squared:                       0.006
Model:                            OLS   Adj. R-squared:                 -0.002
Method:                 Least Squares   F-statistic:                    0.7503
Date:                Mon, 27 Oct 2025   Prob (F-statistic):              0.388
Time:                        13:40:34   Log-Likelihood:                -97.728
No. Observations:                 120   AIC:                             199.5
Df Residuals:                     118   BIC:                             205.0
Df Model:                           1                                         
Covariance Type:            nonrobust                                         
                     coef    std err          t      P>|t|      [0.025      0.975]
----------------------------------------------------------------------------------
Intercept          4.0362      0.072     55.

Based on the regression results (F(1,118) = 0.75, p = 0.388), there is no statistically significant difference in teaching evaluation ratings between male and female instructors.

Although male instructors scored about 0.09 points lower on average than female instructors, this difference is not statistically significant and could be due to random variation rather than a real gender effect.

# Q2. Regression with ANOVA: Using the teachers' rating data set, does beauty score for instructors differ by age?

In [11]:
df['age_group'] = pd.cut(df['Age'], bins=[20, 30, 40, 50, 60, 70],
                         labels=['20s', '30s', '40s', '50s', '60s'])
model = smf.ols('Beauty ~ C(age_group)', data=df).fit()
anova_table = sm.stats.anova_lm(model, typ=2)
print(anova_table)


                 sum_sq     df         F    PR(>F)
C(age_group)   2.294368    4.0  1.038355  0.378383
Residual      64.078891  116.0       NaN       NaN


/usr/local/lib/python3.12/dist-packages/statsmodels/base/model.py:1894: ValueWarning: covariance of constraints does not have full rank. The number of constraints is 4, but rank is 3
  warnings.warn('covariance of constraints does not have full '


An ANOVA test was conducted to determine whether beauty scores differ by age group.
The results were not statistically significant, F(4,116) = 1.03, p = 0.378.
This indicates that instructors’ beauty scores do not significantly differ across age groups.

# Q3. Correlation: Using the teachers' rating dataset, Is teaching evaluation score correlated with beauty score?

In [12]:
import statsmodels.api as sm

X = df['Beauty']
y = df['Evaluation']

X = sm.add_constant(X)

model = sm.OLS(y, X).fit()
predictions = model.predict(X)

print(model.summary())


                            OLS Regression Results                            
Dep. Variable:             Evaluation   R-squared:                       0.009
Model:                            OLS   Adj. R-squared:                  0.001
Method:                 Least Squares   F-statistic:                     1.107
Date:                Mon, 27 Oct 2025   Prob (F-statistic):              0.295
Time:                        13:44:58   Log-Likelihood:                -97.548
No. Observations:                 120   AIC:                             199.1
Df Residuals:                     118   BIC:                             204.7
Df Model:                           1                                         
Covariance Type:            nonrobust                                         
                 coef    std err          t      P>|t|      [0.025      0.975]
------------------------------------------------------------------------------
const          4.2624      0.263     16.228      0.0

[1] Standard Errors assume that the covariance matrix of the errors is correctly specified.

The model was not statistically significant, F(1,118) = 1.11, p = 0.295, with an R² of 0.009.
Beauty score was not a significant predictor of teaching evaluations (β = -0.071, p = 0.295), indicating that instructors’ beauty scores have no meaningful impact on their evaluation ratings in this dataset.